src/System/executable_query.rs

```diff
@@ -12,18 +13,28 @@ use super::save_component::PruningStrategy;
 
 pub enum QueryResult {
     Refinement(bool),
+    Reachability(bool, Vec<Edge>), // This represents a path from start state to end state
     GetComponent(Component),
     Consistency(bool),
     Determinism(bool),
     Error(String),
 }
 
+/// When egde is fully defined in protobuf it will:
+/// Serves as a representation of a transition you can take in the transision system.
+pub struct Edge {
+    id: String,
+}
+
 impl QueryResult {
     pub fn print_result(&self, query_str: &str) {
```

Comment:
Might remove this and use the already existing definition of edge. Otherwise, we'll rename this to a more appropriate name and make sure it is well documented 😄

Reply:
It will be replaced by the real struct for protobuf edge when it is ready, plz resolve =)

Reply:
We have removed this, because it is not used anymore. A suggestion from t.lohse
@Casper-NS